src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -29,154 +29,121 @@ class TaskReplaceActor(
+  @SuppressWarnings(Array("all")) // async/await
+  override def preStart(): Unit = {
+    super.preStart()
+    // subscribe to all needed events
+    eventBus.subscribe(self, classOf[HealthStatusResponse])
 
-  // compute all variables maintained in this actor =========================================================
+    // reset the launch queue delay
+    logger.info("Resetting the backoff delay before restarting the runSpec")
+    launchQueue.resetDelay(runSpec)
 
-  // Only old instances that still have the Goal.Running
-  val oldActiveInstances = oldInstances.filter(_.state.goal == Goal.Running)
+    // Fetch state of apps
+    val system = akka.actor.ActorSystem("system")
+    tick = system.scheduler.schedule(0 seconds, 10 seconds)(request_health_status)
+  }
 
-  // All instances to kill as set for quick lookup
-  private[this] var oldActiveInstanceIds: SortedSet[Id] = oldActiveInstances.map(_.instanceId).to[SortedSet]
+  def isHealthy(instance: Instance, healths: Map[Instance.Id, Seq[Health]]): Boolean = {
+    // FIXME: Or use state.healthy.getOrElse
+    val i_health = healths.find(_._1 == instance.instanceId)
+    if (instance.hasConfiguredHealthChecks && i_health.isDefined) { // Has healthchecks
+      if (_isHealthy(i_health.get._2) == true) {
+        logger.info(s"Checking ${instance.instanceId}: has healcheck and is healthy")
+        return true
+      } else {
+        logger.info(s"Checking ${instance.instanceId}: has healcheck and is NOT healthy")
+        return false
+      }
+    } else if (instance.hasConfiguredHealthChecks && !i_health.isDefined) {
+      logger.info(s"Checking ${instance.instanceId}: has healcheck and is UNKNOWN")
+      return false
+    } else { // Don't have healthchecks
+      logger.info(s"Checking ${instance.instanceId}: has now HC")
+      if (instance.state.goal == Goal.Running) return true
+      else return false
+    }
+  }
 
-  // All instances to kill queued up
-  private[this] val toKill: mutable.Queue[Instance.Id] = oldActiveInstances.map(_.instanceId).to[mutable.Queue]
+  def _isHealthy(healths: Seq[Health]): Boolean = {
+    for (h <- healths)
+      if (!h.alive) return false
+    return true
+  }
 
-  // The number of started instances. Defaults to the number of already started instances.
-  var instancesStarted: Int = instancesAlreadyStarted.size
+  def request_health_status(): Unit = {
+    logger.info(s"Requesting HealthStatus for ${pathId}")
+    deploymentManagerActor ! HealthStatusRequest(pathId)
+  }
 
-  @SuppressWarnings(Array("all")) // async/await
-  override def preStart(): Unit = {
-    super.preStart()
-    // subscribe to all needed events
-    eventBus.subscribe(self, classOf[InstanceChanged])
-    eventBus.subscribe(self, classOf[InstanceHealthChanged])
+  def step(health: Map[Instance.Id, Seq[Health]]): Unit = {
+    logger.info(s"---=== DEPLOYMENT STEP FOR ${pathId} ===---yy")
```

Comment:
move it to debug at some point?

Reply:
Yes, you're right; i'll do it at the end to avoid having to enable debug logs for all actors  :)

Reply:
Can I leave other traces as info? Like state of tasks, and number of tasks to kill/start? I'll merge into one line, but this is something that is currently in upstream code so I think we should keep track of this anyway